File: 53.go
func maxSubArray(nums []int) int {
	ans := nums[0]
	len := len(nums)
	for i := 1; i < len; i++ {
		if nums[i]+nums[i-1] > nums[i] {
			nums[i] = nums[i] + nums[i-1]
		}
		if nums[i] > ans {
			ans = nums[i]
		}
	}
	for _, v := range nums {
		fmt.Println(v)
	}
	return ans
}

//-2,1,-3,4,-1,2,1,-5,4
//-2,1,-2,4,3,5,6,1,5

File: 53.java
class Solution {
    public int maxSubArray(int[] nums) {
        int l = nums.length, ans = nums[0];
        int[] dp = new int[l];
        dp[0] = nums[0];
        for(int i = 1;i < l;i ++) {
            dp[i] = nums[i] + Math.max(dp[i], dp[i-1]);
            ans = Math.max(ans, dp[i]);
        }
        return ans;
    }
}